client/my-sites/domains/domain-management/edit/transfer.jsx

```diff
@@ -43,26 +44,76 @@ class Transfer extends React.PureComponent {
 		this.props.paymentSettingsClick( this.props.domain );
 	};
 
+	restartTransfer = () => {
+		const { domain, selectedSite, translate } = this.props;
+
+		restartInboundTransfer( selectedSite.ID, domain.name, ( error, result ) => {
+			if ( result ) {
+				this.props.successNotice( translate( 'The transfer has been successfully restarted.' ), {
+					duration: 5000,
+				} );
+				fetchDomains( selectedSite.ID );
+			} else {
+				this.props.errorNotice(
+					error.message || translate( 'We were unable to restart the transfer.' ),
+					{
+						duration: 5000,
+					}
+				);
+			}
+		} );
+	};
+
+	getCancelledContent() {
+		const { domain, translate } = this.props;
+
+		return (
+			<Card>
+				<div>
+					<p className="edit__transfer_text_fail">{ translate( 'Domain Transfer Failed' ) }</p>
+					<p>
+						{ translate(
+							'We were unable to verify the transfer of {{strong}}%(domain)s{{/strong}}. The domain ' +
+								'authorization code from you current registrar was not provided to initiate the transfer. ' +
```

Comment:
I wonder if we should put something like "or the transfer was cancelled" here?

Reply:
@danhauk good point for copy here.

Reply:
@aidvu Ideally if we actually _know_ the reason it failed we customize the message based on that. If we can't get that, then adding @hambai's suggestion is a good compromise.

Reply:
@danhauk we should make a general message for now, and see what kind of errors we get, and if they are any good for specific messaging.

Reply:
@aidvu What other kinds of reasons could we have for the transfer failing? Would adding "or the transfer was cancelled" be enough or could there be other reasons?

Reply:
Hmm. Domain relocked. Time passed and was never confirmed (so not really the same as cancelled).